Rust/Rust.sublime-syntax

```diff
@@ -1393,6 +1398,33 @@ contexts:
           scope: punctuation.definition.string.end.rust
           pop: true
 
+  c-string:
+    - match: '(c)(")'
+      captures:
+        1: storage.type.string.rust
+        2: punctuation.definition.string.begin.rust
+      push:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.double.rust
+        - match: '"'
+          scope: punctuation.definition.string.end.rust
+          pop: true
+        - include: escaped-byte
+        - include: line-continuation
+        - include: escaped-char
```

Comment:
Sorry for maybe dumb question. I don't know Rust. Can a double quote be escaped? If so, this should go above the string end (and add regression tests). If not, please disregard.

Reply:
Double quotes can be escaped via `\"` like other strings.

I added some more tests around escaping.

I'm reluctant to rearrange the rules here. I'm uncertain if there was an explicit reason for the order the rules are specified, like if the order would impact performance (I'm not sure how smart the engines are for combining the rules). Logically it shouldn't matter since there isn't any overlap between these particular rules.

Reply:
We always have popping end patterns at the beginning, followed by content such as escapes in strings.

As escape pattern starts with `\` there's no chance of escapes being consumed by end pattern by accident. So order doesn't matter.